portfolio/src/main/java/com/google/sps/data/LongestTaskFirstScheduler.java

```diff
@@ -69,36 +61,147 @@ public SchedulingAlgorithmType getSchedulingAlgorithmType() {
     return SchedulingAlgorithmType.LONGEST_TASK_FIRST;
   }
 
+  /** Constructs a list for the updated available time ranges after deletion. */
+  private List<TimeRange> constructAvailableTimeRanges(TimeRangeGroup updatedTimeRangeGroup) {
+    Iterator<TimeRange> updatedAvailableTimesGroupIterator = updatedTimeRangeGroup.iterator();
```

Comment:
I think you can make this simpler. You should be able to do:
```
List<TimeRange> updatedAvailableTimes = new ArrayList<>();
updatedTimeRangeGroup.iterator().forEachRemaining(updatedAvailableTimes::add);
return updatedAvailableTimes;
```

Reply:
Updated.

Reply:
nit: You can remove line 62 now I think?